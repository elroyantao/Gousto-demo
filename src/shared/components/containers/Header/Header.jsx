import React, { Component } from 'react'
import PropTypes from 'prop-types'
// import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'

export default class Header extends Component {

  renderNavItem(category) {
    return (
      <NavLink
        to={`/category/${category.id}`}
        key={category.id}
        activeStyle={{
          fontWeight: 'bold',
          color: 'red'
        }}
      >
        {category.title}
      </NavLink>
    )
  }

  render() {
    return (
      <nav>
        {this.props.categories.map(this.renderNavItem)}
      </nav>
    )
  }
}

Header.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    box_limit: PropTypes.number,
    hidden: PropTypes.bool,
    id: PropTypes.string,
    is_default: PropTypes.bool,
    recently_added: PropTypes.bool,
    title: PropTypes.string
  })).isRequired
}
