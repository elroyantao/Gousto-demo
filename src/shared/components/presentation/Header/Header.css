.Header {
  border-bottom: solid black 1px;
  padding-bottom: 2px;
}

.Header-title {
  font-size: 30px;
  color: rgb(224, 103, 78);
  display: flex;
  justify-content: center;
  padding: 20px;
  background-color: #FBF9F6;
}

.Header-nav {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  font-size: 20px;
  font-weight: 400;
}

.Header-navItem {
  padding: 5px;
  margin: 3px 2px;
  text-decoration: none;
  color: #645B57;
}

.Header-navItem--active {
  composes: Header-nav;
  color: white;
  background-color: #645B57;
}
