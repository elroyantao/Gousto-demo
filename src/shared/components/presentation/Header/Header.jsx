import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

import styles from './Header.css'

export default class Header extends Component {
  static propTypes = {
    categories: PropTypes.arrayOf(PropTypes.shape({
      box_limit: PropTypes.number,
      hidden: PropTypes.bool,
      id: PropTypes.string,
      is_default: PropTypes.bool,
      recently_added: PropTypes.bool,
      title: PropTypes.string
    }))
  }.isRequired

  static defaultProps = {
    categories: []
  }

  renderNavItem(category) {
    return (
      <NavLink
        to={`/category/${category.id}`}
        key={category.id}
        className={styles['Header-navItem']}
        activeClassName={styles['Header-navItem--active']}
      >
        {category.title}
      </NavLink>
    )
  }

  render() {
    return (
      <section className={styles.Header}>
        <div className={styles['Header-title']}> Gousto </div>
        <nav className={styles['Header-nav']}>
          {this.props.categories.map(this.renderNavItem)}
        </nav>
      </section>
    )
  }
}
