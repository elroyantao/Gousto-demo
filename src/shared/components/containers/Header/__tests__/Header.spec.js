import testComponentHelper from '../../../../lib/test-helper'
import Header from '../Header'

describe('<Header/>', () => {
  const renderComponent = testComponentHelper(Header)
  it('render in default state', () => {
    expect(renderComponent({}).getTree()).toMatchSnapshot()
  })
})
