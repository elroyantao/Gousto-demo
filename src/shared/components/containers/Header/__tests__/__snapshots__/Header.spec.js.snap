// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> render in default state 1`] = `
<nav>
  header
</nav>
`;
